### PR TITLE
[Packages][Lucide] Switch to `data-lucide`

### LINKED ISSUE #981: support HTML standard data-* attributes
Is it possible to add support for data-* attributes so that users who prefer to write standards-compliant code can use Lucide without having to add exceptions to code quality checking tools?

For example, instead of `<span icon-name='box'></span>` it should be possible to write `<span data-icon-name='box'></span>`.

The two options can co-exist if needed to maintain backward compatibility.

https://developer.mozilla.org/en-US/docs/Learn/HTML/Howto/Use_data_attributes

### PR DESCRIPTION
**Pull request by karsa-mistmere**
closes #981

### CODE REVIEW

File: packages/lucide/src/lucide.ts [hunk 1]
```diff
@@ -5,7 +5,7 @@ import * as allIcons from './icons';
  * Replaces all elements with matching nameAttr with the defined icons
  * @param {{ icons?: object, nameAttr?: string, attrs?: object }} options
  */
-const createIcons = ({ icons = {}, nameAttr = 'icon-name', attrs = {} } = {}) => {
```
Comment (ericfennis) [inline packages/lucide/src/lucide.ts:8]: We maybe should consider that we have an overlapping time, with changing this.
So maybe we should support the old attribute as well and console a warning, with saying we remove this in v1?

Comment (karsa-mistmere) [reply packages/lucide/src/lucide.ts:8]: Yeah, good idea, I've posted an implementation of this, also checking if the default attribute name was used.

File: packages/lucide/src/lucide.ts [hunk 2]
```diff
@@ -17,10 +17,20 @@ const createIcons = ({ icons = {}, nameAttr = 'icon-name', attrs = {} } = {}) =>
...
-
   Array.from(elementsToReplace).forEach((element) =>
     replaceElement(element, { nameAttr, icons, attrs }),
   );
+
+  /** @todo: remove this block in v1.0 */
+  if (nameAttr === 'data-lucide') {
+    const deprecatedElements = document.querySelectorAll('[icon-name]');
+    if (deprecatedElements.length > 0) {
+      console.warn('Icons were found with the now deprecated icon-name attribute. These will still be replaced for backwards compatibility, but will no longer be supported in v1.0');
```
Comment (ericfennis) [inline packages/lucide/src/lucide.ts:28]: Maybe mention our package name somewhere. So devs know where this warning is coming from.

Comment (karsa-mistmere) [reply packages/lucide/src/lucide.ts:28]: Oh, sure, also, we should add the new attribute name anyway.

Comment (karsa-mistmere) [reply packages/lucide/src/lucide.ts:28]:
```suggestion
      console.warn('[Lucide] Some icons were found with the now deprecated icon-name attribute. These will still be replaced for backwards compatibility, but will no longer be supported in v1.0 and you should switch to data-lucide');
```

Comment (ericfennis) [reply packages/lucide/src/lucide.ts]: perfecto!

**Pull request merged**